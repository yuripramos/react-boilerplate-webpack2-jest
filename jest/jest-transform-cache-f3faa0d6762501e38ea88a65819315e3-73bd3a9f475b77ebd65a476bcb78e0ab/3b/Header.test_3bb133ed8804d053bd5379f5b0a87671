'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _enzyme = require('enzyme');

var _Header = require('./Header.js');var _Header2 = _interopRequireDefault(_Header);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('rendering tests', function () {
	it('rendering correctly', function () {
		var component = (0, _enzyme.shallow)(_react2.default.createElement(_Header2.default, null));
		expect(component).toMatchSnapshot();
	});
});