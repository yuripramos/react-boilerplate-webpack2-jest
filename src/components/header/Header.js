import React, { Component } from 'react';

import style from './style.scss';

function increaseScore(state) {
  return { score: state.score + 1 };
}
// Functional setState example

class Header extends Component {
  constructor(props) {
    super(props);
    this.handleIncreaseScore = this.handleIncreaseScore.bind(this);
  }

  handleIncreaseScore() {
    this.setState(increaseScore);
  }

  render() {
    return (
      <div className={`${style.container} row`} >
        <div className="col-12">
          <h1> React JW2 boilerplate </h1>
        </div>
      </div>
    );
  }
}

export default Header;
