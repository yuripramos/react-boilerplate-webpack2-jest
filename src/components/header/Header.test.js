import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header.js';

describe('rendering tests', () => {
  it('rendering correctly', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
