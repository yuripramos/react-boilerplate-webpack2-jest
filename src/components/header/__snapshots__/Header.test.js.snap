// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering tests rendering correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="container row"
>
    <div
        className="col-12"
    >
        <h1>
             React JW2 boilerplate 
        </h1>
    </div>
</div>,
  "nodes": Array [
    <div
      className="container row"
>
      <div
            className="col-12"
      >
            <h1>
                   React JW2 boilerplate 
            </h1>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleIncreaseScore": [Function],
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="container row"
>
          <div
                    className="col-12"
          >
                    <h1>
                               React JW2 boilerplate 
                    </h1>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="container row"
>
          <div
                    className="col-12"
          >
                    <h1>
                               React JW2 boilerplate 
                    </h1>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
